src/puya/ir/models.py

```diff
@@ -223,6 +223,15 @@ def accept(self, visitor: IRVisitor[T]) -> T:
         return visitor.visit_biguint_constant(self)
 
 
+@attrs.frozen
+class DeployVar(Value):
```

Comment:
I opted for introducing a new model instead of changing the existing integer/bytes constant types as there's a lot of optimization code which works on these constants which wouldn't apply to deploy time vars

Reply:
That makes sense, but also got me thinking that it might be a nice addition to allow specifying the values for these variables during compilation, a bit like `-D` style macros in c compiler's. If those values are provided then these could become actual constants and would allow optimisation

Reply:
@daniel-makerx  the `-D` feature probably works better if/when we allow `os.getenv` at the module level.

@tristanmenzel can we rename this to `TemplateVar`?